### DISCUSSION TITLE
[apple2] 80 column scroll function for conio?

### DISCUSSION THREAD

**Discussion by tschak909** [cc65/cc65#1982] [Q&A]
What is the best way to get the screen to scroll while using the 80 column card? Is the solution to do the soft-switch dance and call the slot 3 dispatch? Has somebody already written a function to do that?

-Thom

**Comment by polluks** [comment]
You know, conio does not support scrolling at all. Maybe you can do it at high level.
What about stdio?

**Comment by mrdudz** [comment]
There was a discussion about adding scroll functions to conio.... but as so often, it died when certain ppl with strong opinions couldnt agree. One of the many things on my todo list for cc65

**Comment by colinleroy** [comment]
Yeah, for now scrolling, be it 40 or 80 columns, is 
```
bit $C082
jsr $FC70
bit $C080
```

The FC70 rom entry point is stable across various models, and handle 80-columns on models that can.

Sadly, ROM disassemblies show that some models can scroll both ways, but those are internal entry points and differ across models, so if you want to scroll down instead of up, you'll just have to duplicate the code. (obvious disadvantage is code size):
https://github.com/colinleroy/a2tools/blob/master/src/lib/scroll.s
